### PR TITLE
Update Go-Ethereum Fork

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Update Dependency

**What does this PR do? Why is it needed?**

- [x] Updates the version of the bazelified go-ethereum fork that prysm uses.
 
**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -3613,3 +3613,27 @@ def prysm_deps():
         sum = "h1:PeVNzgTRtWGm6fVic5i21t+n5ptPGCZuMcSPVMyTWjs=",
         version = "v0.0.0-20180321022601-755488143dae",
     )
+    go_repository(
+        name = "com_github_cloudflare_cloudflare_go",
+        importpath = "github.com/cloudflare/cloudflare-go",
+        sum = "h1:J82+/8rub3qSy0HxEnoYD8cs+HDlHWYrqYXe2Vqxluk=",
+        version = "v0.10.2-0.20190916151808-a80f83b9add9",
+    )
```
Comment (farazdagi) [inline deps.bzl:3621]: - Update this PR, after https://github.com/prysmaticlabs/prysm/pull/7301 is merged (we need to re-run `bazel run //:gazelle -- update-repos -from_file=go.mod -to_macro=deps.bzl%prysm_deps` and make sure that clougflares roughttime is not included into project - it is removed in https://github.com/prysmaticlabs/prysm/pull/7299)

**Review by farazdagi — CHANGES_REQUESTED** [review]

File: deps.bzl (hunk 1 above)
Comment (nisdas) [reply deps.bzl:3614]: @farazdagi I think that dep is used by geth which is why it has been added. It seems separate from roughtime. I resynced with master and no changes for me locally from running `gazelle`

Comment (farazdagi) [reply deps.bzl:3614]: image

I suppose go-ethereum brings that dependency, Nishant, can you please double check that every updated import is intended?

**Review by farazdagi — APPROVED** [review]
lgtm

**Review by farazdagi — CHANGES_REQUESTED** [review]
cloudflare needs to be added back, as it is used by go-ethereum (see `go mod graph | grep cloudf`)

**Review by farazdagi — APPROVED** [review]

**Pull request merged**